### PR TITLE
Add DLL export symbols on windows platform

### PR DESCRIPTION
**Pull request by jmoguillansky-gpsw**
Rework shared library symbol export logic.  
Symbols are hidden by default.
When building library, declare public APIs as visible in header file.  
When the user includes the public header file, specify the symbol visibility of the public APIs for the linker.

### CODE REVIEW

File: libnodegl/meson.build [hunk 1]
```diff
@@ -360,7 +363,8 @@ libnodegl = library(
   install: true,
   install_rpath: install_rpath,
   version: meson.project_version(),
-  link_args: lib_link_args,
+  c_args: lib_c_args,
+  gnu_symbol_visibility: 'hidden'
```
Comment (mbouron-gpsw) [inline libnodegl/meson.build:367]: Missing coma at the end of the line.

File: libnodegl/meson.build [hunk 2]
```diff
@@ -346,12 +350,11 @@ endforeach
 
 configure_file(output: 'config.h', configuration: conf_data)
 
-# See https://github.com/mesonbuild/meson/pull/4747 to follow advancement on
-# the native support for symbol visibility
-lib_link_args = cc.get_supported_link_arguments([
-  '-Wl,--version-script,@0@'.format(meson.current_source_dir() / 'libnodegl.symexport'),  # GNU ld
-  '-Wl,-exported_symbols_list,@0@'.format(meson.current_source_dir() / 'libnodegl.darwin.symexport'),  # Darwin
-])
+lib_c_args = ''
```
Comment (mbouron-gpsw) [inline libnodegl/meson.build:353]: lib_c_args should be a list

File: libnodegl/meson.build [hunk 3]
```diff
@@ -346,12 +350,11 @@ endforeach
...
+
```
Comment (mbouron-gpsw) [inline libnodegl/meson.build:354]: Please remove extra linebreak.

File: libnodegl/meson.build [hunk 4]
```diff
@@ -370,10 +374,22 @@ lib_header = configure_file(
 install_headers(lib_header)
 
 pkg = import('pkgconfig')
+pkg_extra_cflags = []
+
+# pkg-config extra cflags don't apply to local executables, so we have to
+# transfer them manually. Alternatively, we could create a virtual
+# dependency and use it.
+# See https://github.com/mesonbuild/meson/issues/8082 for more information
+
```
Comment (mbouron-gpsw) [inline libnodegl/meson.build:383]: Please remove extra linebreak.

File: libnodegl/meson.build [hunk 5]
```diff
@@ -370,10 +374,22 @@ lib_header = configure_file(
 install_headers(lib_header)
 
 pkg = import('pkgconfig')
+pkg_extra_cflags = []
+
+# pkg-config extra cflags don't apply to local executables, so we have to
+# transfer them manually. Alternatively, we could create a virtual
+# dependency and use it.
+# See https://github.com/mesonbuild/meson/issues/8082 for more information
```
Comment (mbouron-gpsw) [inline libnodegl/meson.build:382]: This comment does not match what is done below, it belongs to the executables configuration not the library itself.

File: libnodegl/meson.build [hunk 6]
```diff
@@ -295,7 +299,7 @@ gbackends_cfg = {
       'windows': {
         'src': files('backends/gl/glcontext_wgl.c'),
         'cfg': 'HAVE_GLPLATFORM_WGL',
-        'libs': ['opengl32', 'gdi32'],
+        'libs': ['OpenGL32', 'gdi32'],
```
Comment (mbouron-gpsw) [inline libnodegl/meson.build:302]: This belongs in a separate commit.

Comment (jmoguillansky-gpsw) [reply libnodegl/meson.build:302]: updated

File: libnodegl/meson.build (hunk 1 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/meson.build]: updated

File: libnodegl/meson.build (hunk 2 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/meson.build]: updated

File: libnodegl/meson.build (hunk 3 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/meson.build]: updated

File: libnodegl/meson.build (hunk 4 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/meson.build]: updated

File: libnodegl/meson.build (hunk 5 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/meson.build]: updated.  I think we actually don't need to set the extra nodegl cflags in the local test executable configuration because it's not linking against nodegl, it's just unit test code

**Comment by mbouron-gpsw** [comment]
Can you adjust the commit message so it matches the one in sxplayer ?

**Comment by jmoguillansky-gpsw** [comment]
updated
